rpc/client/options_test.go

```diff
@@ -0,0 +1,158 @@
+package client
+
+import (
+	"crypto/tls"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestWithNetworkURIAddress(t *testing.T) {
+	testCases := []struct {
+		uri string
+
+		wantHost string
+		wantTLS  bool
+	}{
+		{
+			uri:      "grpc://neofs.example.com:8080",
+			wantHost: "neofs.example.com:8080",
+			wantTLS:  false,
+		},
+		{
+			uri:      "grpcs://neofs.example.com:8080",
+			wantHost: "neofs.example.com:8080",
+			wantTLS:  true,
+		},
+		{
+			uri:      "wrongScheme://neofs.example.com:8080",
+			wantHost: "",
+			wantTLS:  false,
+		},
+		{
+			uri:      "impossibleToParseIt",
+			wantHost: "",
+			wantTLS:  false,
+		},
```

Comment:
I think there should be one more case, otherwise it won't be backward compatible with existing applications like Neo oracle node.
```go
{
	uri:      "neofs.example.com:8080",
	wantHost: "neofs.example.com:8080",
	wantTLS:  false,
}
```

To make this test pass, you can check if host address is empty. Maybe you'll find better solution for this, but as I saw, such URI parsed correctly as scheme in `url.ParseRequestURI`, therefore host address is empty.
```go
// options.go
if len(uri.Hostname()) == 0 {
	c.addr = v
	return
}

switch uri.Scheme {
...
```

Reply:
We can leave it as it is, though, but this will require for mandatory use of `grpc://` and `grpcs://`. Maybe it is not that bad, but I would prefer to have "default" scheme for such cases.

Reply:
added such test case and added case: `if URI.Opaque != nil` than suppose that it is address in `host:port` format